src/antsibull_build/build_ansible_commands.py

```diff
@@ -435,6 +437,57 @@ def validate_deps_data(
                 )
 
 
+def check_for_regressions(
+    ansible_version: PypiVer, deps: DependencyFileData, data_dir: os.PathLike[str] | str
+) -> None:
+    dependencies = {
+        PypiVer(deps_ansible_version): deps
+        for deps_ansible_version, deps in load_all_dependency_files(
+            data_dir,
+            accept_deps_file=lambda path, deps_ansible_version: PypiVer(
+                deps_ansible_version
+            )
+            < ansible_version,
+        ).items()
+    }
+    if not dependencies:
+        return
+    previous_release = max(dependencies)
+    previous_release_deps = dependencies[previous_release]
+
+    _VT = t.TypeVar("_VT", SemVer, PypiVer)
+
+    def _check_for_regression(
+        what: str,
+        new_version: str,
+        old_version: str,
+        version_constructor: t.Callable[[str], _VT],
+    ) -> None:
+        new_ver = version_constructor(new_version)
+        old_ver = version_constructor(old_version)
+        if old_ver > new_ver:
+            raise ValueError(
+                f"The old version ({old_version}) of {what}"
+                f" included in Ansible {previous_release} is newer"
+                f" than the new version ({new_version})"
+                f" that would be included in Ansible {ansible_version}!"
```

Comment:
I would probably get rid of "old" and "new", as I think it could be confusing since the version that's "old" is newer than the version that's "new" in the case of a version regression.

```suggestion
                f"The version ({old_version}) of {what}"
                f" included in Ansible {previous_release} is newer"
                f" than the version ({new_version})"
                f" that would be included in Ansible {ansible_version}!"
```

Reply:
How about replacing 'old' by 'previous' and removing 'new'?
```suggestion
                f"The previous version ({old_version}) of {what}"
                f" included in Ansible {previous_release} is newer"
                f" than the version ({new_version})"
                f" that would be included in Ansible {ansible_version}!"
```